src/main/java/de/tum/in/www1/artemis/repository/BuildLogEntryRepository.java

```diff
@@ -11,4 +14,17 @@
 @Repository
 public interface BuildLogEntryRepository extends JpaRepository<BuildLogEntry, Long> {
 
+    /**
+     * deletes all BuildLogEntry s of a ProgrammingSubmission
+     *
+     * @param programmingSubmissionId the id of the submission
+     */
+    @Modifying
+    @Query("""
+            DELETE
+            FROM BuildLogEntry b
+            WHERE b.programmingSubmission.id = :programmingSubmissionId
+            """)
+    void deleteAllByProgrammingSubmissionId(@Param("programmingSubmissionId") Long programmingSubmissionId);
```

Comment:
I think you can simply write 
`void deleteAllByProgrammingSubmission(ProgrammingSubmission programmingSubmission);`

There is no need to write a custom query